docs/spacing.md

```diff
@@ -0,0 +1,108 @@
+# Spacing
+
+Careful spacing is key for the user to understand connections and hierarchies between elements, and for an interface to feel pleasant.
+
+## Scale
+
+Primer follows a base-8 scale, as it allows for significant flexibility when defining a variety of spacing values within an interface. 
+
+The possible values are: 0, 4, 8, 16, 24, 32, 40, 48
+
+For very small components where 8 pixel spacing is too large, you can use 4 pixels. 
+
+## Responsive breakpoints
+
+You can define different spacing values for different breakpoints, so the interface can better adapt to the available space.
+
+`[Example: component with larger margin around it in large screen, vs smaller margin in small screen]`
+
+In code, margins can be defined per breakpoint, where different breakpoints have a different margin applied.
+
+`[Example: mb-sm-2 mb-md-3 mb-lg-4]`
+
+## Applying spacing
+
+The general rule when using Primer is to apply spacing in increments of 8 pixels, as provided by the spacing utilities. 
+
+In some cases, for very small components and in denser areas of the UI, you can use 4 pixels.
+
+`[Example: text and icon button]`
+
+Primer components have padding built-in, so in most cases you’ll only need to consider vertical spacing *between* components.
+
+In some cases, the spacing values inside components don't follow the values in the scale, but the total size of the component should always follow the scale.
+
+`[Example: the padding inside a button may be 3px or 5px but the total height is 32px]`
```

Comment:
Todo: add image of a component with irregular padding values where total height follows the scale.

Reply:
@ashygee use large button as example